### PR TITLE
Add DILabel functions for LLVM-C

### PR DESCRIPTION
**Pull request by tf2spi**
Addresses #112799

### CODE REVIEW

File: llvm/include/llvm-c/DebugInfo.h [hunk 1]
```diff
@@ -1415,6 +1415,52 @@ LLVMMetadataRef LLVMInstructionGetDebugLoc(LLVMValueRef Inst);
...
+/**
+ * Create a new descriptor for a label
+ *
+ * \param Builder         The DIBuilder.
+ * \param Scope           The scope to create the label in.
+ * \param Name            Variable name.
+ * \param NameLen         Length of variable name.
+ * \param File            The file to create the label in.
+ * \param LineNo          Line Number.
+ * \param AlwaysPreserve  Preserve the label even if the optimizer wants to remove it.
```
Comment (tf2spi) [inline llvm/include/llvm-c/DebugInfo.h:1427]: I noticed after the fact that this line is longer than 80 characters and that the rest of the lines in this file are 80 characters. I would make a change like this but I also don't want to run a pipeline right now for a simple comment change. Waiting for review then, if this is desired, I can add it in. 
```suggestion
 * \param AlwaysPreserve  Preserve the label regardless of optimization.
```

**Comment by tf2spi** [comment]
I also noticed there's an [LLVM-C DebugInfo](https://github.com/llvm/llvm-project/blob/main/llvm/tools/llvm-c-test/debuginfo.c) test. Do tests for new features normally get added in the same PR or in a separate one?

**Comment by pogo59** [comment]
Tests should be added in the same PR as the functional change.

**Comment by tf2spi** [comment]
@echristo Hello! I'm a new contributor in LLVM. After reading the [LLVM Contributing Guide](https://llvm.org/docs/Contributing.html#how-to-submit-a-patch), it suggested to ping you when the code is ready to review. Any time that's convenient for you to review this I would greatly appreciate.

**Comment by tf2spi** [comment]
Ping

**Comment by tf2spi** [comment]
@llvm/pr-subscribers-debuginfo
@llvm/pr-subscribers-llvm-ir

Ping

**Pull request merged**